### PR TITLE
block collision detection fixes

### PR DESCRIPTION
**Pull request by Daniel6**
resolving #9

### CODE REVIEW

File: src/controller.h [hunk 1]
```diff
@@ -14,7 +14,7 @@
 #define US 1000000 // 1 million microseconds per second
 #define TICK_RATE 60 // Steps per second
 #define DELAY US/TICK_RATE // Delay in microseconds to achieve tick rate
-#define MOVE_RATE 12 // Steps per movement
+#define MOVE_RATE 6 // Steps per movement
```
Comment (thecardkid) [inline src/controller.h:17]: Can you add a todo here to make this non-constant? Eventually we'd want a thing that speeds up right?

Comment (thecardkid) [reply src/controller.h:17]: Assign anyone I guess?

File: src/controller.c [hunk 2]
```diff
@@ -26,11 +26,19 @@ int move_block(int grid[GRID_W][GRID_H], Block *block, int delta_x) {
...
+        // Resolve lateral movement before checking vertical movement
         for (int i = 0; i < 4; i++) {
             int x = block->cells[i][0] + block->x;
             int y = block->cells[i][1] + block->y;
 
-            if (is_within_grid(block->x, delta_x)) delta_x = 0;
+            if (is_within_grid(x, delta_x)) delta_x = 0;
+        }
+
+        // Resolve vertical movement
```
Comment (thecardkid) [inline src/controller.c:38]: Not sure what this is resolving, can you explain?

Comment (thecardkid) [reply src/controller.c:38]: JK I saw the issue

**Review by thecardkid — APPROVED** [review]

File: src/controller.h (hunk 1 above)
Comment (Daniel6) [reply src/controller.h:17]: I'll make a card on Trello(tm)

**Pull request merged**